src/models/inbound/dfspLinking.model.ts

```diff
@@ -157,21 +196,95 @@ export class DFSPLinkingModel
+    // a PATCH /consentRequests/{ID} response containing an authToken
+    const waitOnAuthToken = deferredJob(this.pubSub, waitOnAuthTokenFromPISPResponseChannel)
+      .init(async (channel) => {
+        const res = await this.thirdpartyRequests.putConsentRequests(
+          consentRequestsPostRequest.consentRequestId,
+          consentRequestsIDPutRequest,
+          toParticipantId
+        )
+
+        this.logger.push({ res, channel })
+          .log('ThirdpartyRequests.putConsentRequests call sent to auth service, listening on response')
+      })
+      .job(async (message: Message): Promise<void> => {
+        try {
+          type PutResponse =
+            tpAPI.Schemas.ConsentRequestsIDPatchRequest
+          type PutResponseOrError = PutResponse & fspiopAPI.Schemas.ErrorInformationObject
+          const putResponse = message as unknown as PutResponseOrError
+
+          if (putResponse.errorInformation) {
+            // if the PISP sends back any error, both machines will now
+            // need to be in an errored state
+            // store the error so we can transition to an errored state
+            this.data.errorInformation = putResponse.errorInformation as unknown as fspiopAPI.Schemas.ErrorInformation
+
+          } else {
+            this.data.consentRequestsIDPatchResponse = { ...message as unknown as PutResponse }
+          }
+        } catch (error) {
+          this.logger.push(error).error('ThirdpartyRequests.putConsentRequests request error')
+          return Promise.reject(error)
+        }
+      })
+      // since the PATCH /consentRequest/{ID} PISP response requires user input
+      // on the PISP side we need to make this a longer time.
+      // todo: figure out what time is adequate and make a new variable to store it
+      .wait(this.config.requestProcessingTimeoutSeconds * 1000)
+
+    // catch any unplanned errors and notify PISP
+    try {
+      await waitOnAuthToken
+    } catch (error) {
+      // we send back an account linking error despite the actual error
+      const mojaloopError = reformatError(
+        Errors.MojaloopApiErrorCodes.TP_ACCOUNT_LINKING_ERROR,
+        this.logger
+      )
+
+      // if the flow fails to run for any reason notify the PISP that the account
+      // linking process has failed
+      await this.thirdpartyRequests.putConsentRequestsError(
+        consentRequestsPostRequest.consentRequestId,
+        mojaloopError as unknown as fspiopAPI.Schemas.ErrorInformationObject,
+        this.data.toParticipantId
+      )
+
+      // throw the actual error
+      throw error
+    }
+  }
+
   async onValidateAuthToken (): Promise<void> {
-    const { consentRequestId, consentRequestsIDPatchRequest, toParticipantId } = this.data
+    const { consentRequestId, consentRequestsIDPatchResponse, toParticipantId } = this.data
 
     try {
       const isValidOTP = await this.dfspBackendRequests.validateOTPSecret(
```

Comment:
is there only OTP flow?

Reply:
no. a pull over from the initial code. added a todo to rename instances of the word "OTP" with "authToken".

I didn't address it since there was more pressing matters.